packages/remark-lint-no-shell-dollars/index.js

```diff
@@ -81,11 +80,13 @@ function noShellDollars(tree, file) {
 
     // Check both known shell code and unknown code.
     if (!generated(node) && node.lang && flags.indexOf(node.lang) !== -1) {
-      lines = node.value.split('\n')
+      lines = node.value.split('\n').filter(function (line) {
+        return line
+      })
       length = lines.length
       index = -1
 
-      if (length <= 1) {
+      if (length === 0) {
```

Comment:
Why did this need to change?

Reply:
This is main change: https://github.com/remarkjs/remark-lint/issues/230#issuecomment-616097974

Reply:
Gotcha, stylistic thought, feel free to disagree.
I tend to prefer boundary checks like `length < 1` over equality checks `length === 0`

Reply:
> Gotcha, stylistic thought, feel free to disagree.
> I tend to prefer boundary checks like `length < 1` over equality checks `length === 0`

Why not `length <= 0`? It's difficult to argument. I guess we should not cover negative numbers (undefined behavior).

Reply:
`length <= 0` is also a boundary check.
`<` is ever so slightly more concise, but both can work.

> I guess we should not cover negative numbers

Why not?
It it's more concise, matches more with the previous logic (also a boundary check), and doesn't add complexity.

Reply:
> Why not?

Please, read the following text in the parentheses. You've cut the answer from the quote.

> It it's more concise

What? What do you mean? Like "it's 2 chars shorter"? I don't think it matters.

> matches more with the previous logic (also a boundary check)

Please, describe (with examples if possible).

> and doesn't add complexity.

But it covers unexpected cases! One more time: when a negative value of `length` is possible? Can I add it to tests? No? So let's leave it.

Reply:
Also I mean that `=== 0` (I personally prefer `== 0`) more readable for me, like "it equals zero". In Ruby language there is `length.zero?`. But `< 1` pushes me to thoughts like "wait, is negative value of `length` possible?". The correct answer is "no", as I know, so why we should do that.

Reply:
> > matches more with the previous logic (also a boundary check)
>
> Please, describe (with examples if possible).

`<=` is a boundary check, looking at the diff, the previous code was:
```javascript
if (length <= 1) {
```

> But it covers unexpected cases! One more time: when a negative value of `length` is possible?
> The correct answer is "no", as I know, so why we should do that.

It is unlikely, but possible.
It would 

> But `< 1` pushes me to thoughts like "wait, is negative value of length possible?"

An alternative perspective, coming from more a C background.
It's better to do a boundary check when possible, to prevent the possibility of an underflow.

---

@wooorm any preference on style here?

Reply:
I think this, `=== 0`, is fine!

Reply:
I'm going to resolve this, thank you all. Any other discussions are welcome for me.